### DISCUSSION TITLE
Build Error on windows

### DISCUSSION THREAD

**Discussion by simar1998** [OrcaSlicer/OrcaSlicer#4105] [Dev]
Hey guys, I am trying to build this project on windows 11 and I am running into issues with the bat file throwing errors.  What annoys me is that this worked previously without any issues however now it seems to be not compiling even though its a fresh source download. 

-- Adding precompiled header pchheader.hpp to target libslic3r_gui.
set out_dir to CMAKE_CURRENT_BINARY_DIR/config: C:/Code/C3PD-OS/OrcaSlicer-main/build/src/Release
CMake Error at CMakeLists.txt:641 (file):
  file COPY cannot find
  "C:/Code/C3PD-OS/OrcaSlicer-main/deps/build/OrcaSlicer_dep/usr/local/bin/occt/TKBO.dll":
  File exists.
Call Stack (most recent call first):
  src/CMakeLists.txt:210 (orcaslicer_copy_dlls)

-- libslic3r-CMAKE_BUILD_TYPE: Release
-- CMAKE_CURRENT_BINARY_DIR: C:/Code/C3PD-OS/OrcaSlicer-main/build/src
-- Configuring incomplete, errors occurred!

C:\Code\C3PD-OS\OrcaSlicer-main\build>cmake --build . --config Release --target ALL_BUILD -- -m
Microsoft (R) Build Engine version 16.11.2+f32259642 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1009: Project file does not exist.
Switch: ALL_BUILD.vcxproj

C:\Code\C3PD-OS\OrcaSlicer-main\build>cd ..

C:\Code\C3PD-OS\OrcaSlicer-main>call run_gettext.bat
C:\Code\C3PD-OS\OrcaSlicer-main
Microsoft (R) Build Engine version 16.11.2+f32259642 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1009: Project file does not exist.
Switch: install.vcxproj

**Comment by simar1998** [comment]
I have all the dependencies installed, as stated previously it generated fine previously.

**Comment by stuomas** [comment]
Have you deleted old `build` and `deps/build` directories?

**Reply by simar1998** [reply]
Yes I have, however the error persists unfortunately. Here is the end of the output message of the build_release.bat file 

`CMake Error at C:/Program Files/CMake/share/cmake-3.29/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find wxWidgets (missing: wxWidgets_LIBRARIES
  wxWidgets_INCLUDE_DIRS html adv gl core base webview aui net media)
  (Required is at least version "3.1")
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.29/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  C:/Program Files/CMake/share/cmake-3.29/Modules/FindwxWidgets.cmake:1014 (find_package_handle_standard_args)
  src/CMakeLists.txt:51 (find_package)


-- Configuring incomplete, errors occurred!

C:\Code\C3PD-OS\OrcaSlicer-main\build>cmake --build . --config Release --target ALL_BUILD -- -m
Microsoft (R) Build Engine version 16.11.2+f32259642 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1009: Project file does not exist.
Switch: ALL_BUILD.vcxproj

C:\Code\C3PD-OS\OrcaSlicer-main\build>cd ..

C:\Code\C3PD-OS\OrcaSlicer-main>call run_gettext.bat
C:\Code\C3PD-OS\OrcaSlicer-main
Microsoft (R) Build Engine version 16.11.2+f32259642 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1009: Project file does not exist.
Switch: install.vcxproj`

**Comment by stuomas** [comment]
It's different error though, so progress in a way :)

So are you building deps or using the pre-built ones? (you mentioned you have all deps installed)

**Reply by simar1998** [reply]
Unfortunately no progress has been made. I have no idea how it worked last time. I am going to create a fresh windows vm and see if that helps at this point.

**Reply by stuomas** [reply]
That very much sounds like you haven't deleted the two build directories and then git pull, but I believe you when you said so :)

**Reply by simar1998** [reply]
Thank you for your comment, so I delete the entire deps dir and the build ? Right now I only delete deps/build and the build dir in the root directory. I am assuming that the git pull is to redownload the deleted deps files.

**Reply by simar1998** [reply]
Although a repo from the release gives errors when the build_release.bat file is run. Would you have any suggestions on how to set this project up in a dev environment ? I am kinda new to c++.

**Reply by simar1998** [reply]
I have made some progress, now I am getting the following error. @stuomas 

set out_dir to CMAKE_CURRENT_BINARY_DIR/config: D:/OrcaSlicer-1.9.1/build/src/Release
CMake Error at CMakeLists.txt:641 (file):
  file COPY cannot find
  "D:/OrcaSlicer-1.9.1/deps/build/OrcaSlicer_dep/usr/local/bin/occt/TKBO.dll":
  File exists.
Call Stack (most recent call first):
  src/CMakeLists.txt:210 (orcaslicer_copy_dlls)

**Comment by simar1998** [comment]
SOLVED, the issue was solved for me by Xero, I had additional dependencies installed which were causing issues with cmake. I had added open cascade, wxwidgets etc. after uninstall the bat file with make commands worked fine!

**Comment by vgdh** [comment]
have the same problem, can't compile on win10

**Comment by vgdh** [comment]
-- IGL NOT found, using bundled version...
-- [MCUT] version: 1.2.0
-- [MCUT] MCUT_INCLUDE_DIR=C:/tmp/OrcaSlicer/src/mcut/include
-- [MCUT] compilation_flags=/W4;/wd26812;/wd4131;/wd4704;/wd4245;/wd4297;/wd4505;/wd4701;/bigobj
-- [MCUT] preprocessor_defs=-DMCUT_WITH_COMPUTE_HELPER_THREADPOOL=1;-D_CRT_SECURE_NO_WARNINGS
-- [MCUT] extra_libs=Threads::Threads
-- [MCUT] create target: name=mcut type=STATIC
-- [MCUT] MCUT_LIB_PATH=$<TARGET_FILE:mcut>
CMake Deprecation Warning at src/qoi/CMakeLists.txt:3 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Warning at deps/build/OrcaSlicer_dep/usr/local/lib/cmake/CGAL/CGALConfig.cmake:92 (message):
  CGAL_DATA_DIR cannot be deduced, set the variable CGAL_DATA_DIR to set the
  default value of CGAL::data_file_path()
Call Stack (most recent call first):
  src/libslic3r/CMakeLists.txt:478 (find_package)


-- Visual Leak Detector (VLD) is not found.
-- Using header-only CGAL
-- Targetting Visual Studio 17 2022
-- Target build environment supports auto-linking
-- Using VC toolset 143.
-- Generator uses intermediate configuration directory: $(Configuration)
-- Found Boost: C:/tmp/OrcaSlicer/deps/build/OrcaSlicer_dep/usr/local/lib/cmake/Boost-1.84.0/BoostConfig.cmake (found suitable version "1.84.0", minimum required is "1.48")
-- Boost include dirs: C:/tmp/OrcaSlicer/deps/build/OrcaSlicer_dep/usr/local/include/boost-1_84
-- Boost libraries:
-- OpenCV ARCH: x64
-- OpenCV RUNTIME: vc17
-- OpenCV STATIC: ON
-- Found OpenCV 4.6.0 in C:/tmp/OrcaSlicer/deps/build/OrcaSlicer_dep/usr/local/x64/vc17/staticlib
-- Adding precompiled header pchheader.hpp to target libslic3r.
CMake Deprecation Warning at src/imgui/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Deprecation Warning at src/imguizmo/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- WXWIN environment set to:
CMake Error at C:/Strawberry/c/share/cmake-3.29/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find wxWidgets (missing: wxWidgets_LIBRARIES
  wxWidgets_INCLUDE_DIRS html adv gl core base webview aui net media)
  (Required is at least version "3.1")
Call Stack (most recent call first):
  C:/Strawberry/c/share/cmake-3.29/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  C:/Strawberry/c/share/cmake-3.29/Modules/FindwxWidgets.cmake:1014 (find_package_handle_standard_args)
  src/CMakeLists.txt:52 (find_package)


-- Configuring incomplete, errors occurred!
